test/site/test_curse.py

```diff
@@ -1,16 +1,44 @@
 import unittest
+from dataclasses import dataclass
+from typing import Collection
+from unittest.mock import MagicMock, patch
 
 from updater.site import curse
-from updater.site.enum import GameVersion
+from updater.site.enum import AddonVersion, GameVersion
+from test.testutils import get_file
+
+
+@dataclass
+class VersionTestData:
+    url: str
+    version_regex: str
+    supported_game_versions: Collection[GameVersion]
+
+
+ALL_VERSIONS = (GameVersion.classic, GameVersion.retail, GameVersion.agnostic)
 
 version_test_data = [
-    ['https://www.curseforge.com/wow/addons/classiccodex', r'[0-9]+\.[0-9]+\.[0-9]+'],
-    ['https://www.curseforge.com/wow/addons/bartender4', r"[0-9]+\.[0-9]+\.[0-9]+"],
-    ['https://www.curseforge.com/wow/addons/big-wigs', r"v[0-9]+"],
-    ['https://www.curseforge.com/wow/addons/deadly-boss-mods', r"[0-9]+\.[0-9]+\.[0-9]+"],
-    ['https://www.curseforge.com/wow/addons/weakauras-2', r"[0-9]+\.[0-9]+\.[0-9]+"]
+    VersionTestData(url='https://www.curseforge.com/wow/addons/classiccodex',
+                    version_regex=r'[0-9]+\.[0-9]+\.[0-9]+',
+                    supported_game_versions=(GameVersion.classic, GameVersion.agnostic)),
+    VersionTestData(url='https://www.curseforge.com/wow/addons/bartender4',
+                    version_regex=r"[0-9]+\.[0-9]+\.[0-9]+",
+                    supported_game_versions=ALL_VERSIONS),
+    VersionTestData(url='https://www.curseforge.com/wow/addons/big-wigs',
+                    version_regex=r"v[0-9]+",
+                    supported_game_versions=ALL_VERSIONS),
+    VersionTestData(url='https://www.curseforge.com/wow/addons/deadly-boss-mods',
+                    version_regex=r"[0-9]+\.[0-9]+\.[0-9]+",
+                    supported_game_versions=ALL_VERSIONS),
+    VersionTestData(url='https://www.curseforge.com/wow/addons/weakauras-2',
+                    version_regex=r"[0-9]+\.[0-9]+\.[0-9]+",
+                    supported_game_versions=ALL_VERSIONS)
 ]
 
+MOCK_VERSION_PAGE = MagicMock()
+with open(get_file('mock-curse-version-addons-simc.html'), 'r') as f:
+    MOCK_VERSION_PAGE.text = f.read()
```

Comment:
I totally agree on the greater point about integration testing with the site itself. The reason I set up _this_ particular static test was to test the `AddonType` ordering. As addons are continuously putting out new versions, it was impossible to consistently test the situation where an addon had alpha, beta, and release version on the same page in that respective order. 

Since the other unit tests should fail if the page format changes, this one more serves as "given the _expected_ html, test that the html -> structured format works properly"

I'm open to modifying how the tests work and there is probably a cleaner way to do so, but there are definitely limitations to testing different edge cases on live data. Happy to explore any suggestions you have!

Reply:
That's totally reasonable to me. Love the explanation. Thanks.